src/sentry/static/sentry/app/bootstrap.tsx

```diff
@@ -139,6 +139,34 @@ const render = (Component: React.ComponentType) => {
   }
 };
 
+function changeFavicon(theme: 'dark' | 'light'): void {
+  const n = document.querySelector<HTMLLinkElement>('[rel="icon"][type="image/png"]');
+  if (!n) {
+    return;
+  }
+
+  const path = n.href.split('/sentry/')[0];
+  if (theme === 'dark') {
+    n.href = `${path}/sentry/images/favicon-dark.png`;
+  } else {
+    n.href = `${path}/sentry/images/favicon.png`;
+  }
```

Comment:
personal pref, but I think ternary is a bit easier to read here

```suggestion
  n.href = `${path}/sentry/images/favicon${theme === 'dark' ? '-dark' : ''}.png`;
```

Reply:
or

```suggestion
  n.href = `${path}/sentry/images/${theme === 'dark' ? 'favicon-dark' : 'favicon'}.png`;
```